stan/math/rev/core/vari.hpp

```diff
@@ -450,14 +407,8 @@ class vari_value<T, std::enable_if_t<is_eigen_sparse_base<T>::value>>
    * @tparam S A sparse Eigen type that is convertible to `value_type`
    * @param x Value of the constructed variable.
    */
-  template <typename S, require_convertible_t<S&, value_type>* = nullptr>
-  explicit vari_value(S&& x)
-      : rows_(x.rows()),
-        cols_(x.cols()),
-        size_(x.size()),
-        val_(x),
-        adj_(x),
-        chainable_alloc() {
+  template <typename S, require_convertible_t<S&, T>* = nullptr>
+  explicit vari_value(S&& x) : val_(x), adj_(x), chainable_alloc() {
```

Comment:
```suggestion
  explicit vari_value(const S& x) : val_(x), adj_(x), chainable_alloc() {
```

Reply:
Actually should these be forwarding because sparse holds an actual sparse matrix while the dense matrix one should use `const S&` since it always allocated memory?

Reply:
I changed them all to be `const S&`

Reply:
I thought sparse vari used maps too. Than you are right and forwarding can be taken advantage of. In that case you also need to call forward on the assignment.

Reply:
Done deal!